src/components/common/LoadMoreButton/index.tsx

```diff
@@ -1,9 +1,45 @@
-import { type ReactElement } from 'react'
+import { type MutableRefObject, useCallback, useEffect, useRef, useState, type ReactElement } from 'react'
 import { Button } from '@mui/material'
 
-const LoadMoreButton = ({ onLoadMore }: { onLoadMore: () => void }): ReactElement => {
+const useIntersectionObserver = (element: MutableRefObject<HTMLElement | null>): boolean => {
+  const [isIntersecting, setIsIntersecting] = useState<boolean>(false)
+  const observer = useRef<IntersectionObserver | undefined>()
+
+  const callback = useCallback(([entry]: IntersectionObserverEntry[]) => {
+    setIsIntersecting(entry.isIntersecting)
+  }, [])
+
+  useEffect(() => {
+    observer.current?.disconnect()
+    observer.current = undefined
```

Comment:
You `disconnect` and empty `observer.current` in the cleanup. I don't think this is necessary.